### ISSUE TITLE
Nullability annotations in InlineDataAttribute cause warning for single null value

### ISSUE THREAD

**Issue by rabuckley** [xunit/xunit#3037]
For a Theory method accepting a single nullable parameter, passing `null` to the InlineDataAttribute causes a warning,

```cs
using Xunit;

public sealed class Tests
{
    [Theory]
    [InlineData(null)] // Cannot convert null literal to non-nullable reference type. [CS8625]
    [InlineData("string")]
    public void Test(string? nullableStr)
    {
    }
}
```

Yet, this is valid and works, and the code already handles the case of a `null` array.

https://github.com/xunit/xunit/blob/414e2c88661d9122f1c3e9b595839647a3188119/src/xunit.v3.core/InlineDataAttribute.cs#L20-L22

I can explicitly cast the value to `(string?)` in the attribute, but could the constructor not be changed to take `params object?[]?` to avoid the warning all together, with the `?? = [null]` in the TheoryDataRow constructor?

https://github.com/xunit/xunit/blob/414e2c88661d9122f1c3e9b595839647a3188119/src/xunit.v3.core/InlineDataAttribute.cs#L32-L33

**Comment by rabuckley** [comment]
Thinking about it some more, this would allow `null` to be passed to a method that doesn't accept it. However, xUnit1012 would warn about that case, so not sure...

**Comment by bradwilson** [comment]
Available in v3 `0.5.0-pre.12` https://xunit.net/docs/using-ci-builds
